Documentation/proposals/accepted/202405-agent-daemonset.md

```diff
@@ -84,6 +85,97 @@ The current [PrometheusAgent CRD](https://prometheus-operator.dev/docs/platform/
 
 We will add a new `mode` field that accepts either `StatefulSet` or `DaemonSet`, with `StatefulSet` being the default. If the DaemonSet mode is activated (`mode: DaemonSet`), all the unrelated fields listed above will not be accepted. In the MVP, we will simply fail the reconciliation if any of those fields are set. We will prevent users to directly switch from a live StatefulSet setup to DaemonSet, because that might break their workload if they forget to unset the unsupported fields. Following up, we will leverage validation rules with [Kubernetes' Common Expression Language (CEL)](https://kubernetes.io/docs/reference/using-api/cel/). Only then, we will allow switching from a live StatefulSet setup to DaemonSet. We already have an issue for CEL [here](https://github.com/prometheus-operator/prometheus-operator/issues/5079).
 
+#### 6.1.1 CEL Validation rules
+
+When `mode:DaemonSet`, the following CEL rules will be applied:
+
+- `replicas`
+- `storage`
+- `shards`
+- `persistentVolumeClaimRetentionPolicy`
+- `scrapeConfigSelector`
+- `scrapeConfigNamespaceSelector`
+- `probeSelector`
+- `probeNamespaceSelector`
+- `serviceMonitorSelector`
```

Comment:
since we have added a secondary goal of exploring servicemonitor with endpointslice I am a bit confused about adding this here. Or should that be a different proposal altogether to avoid confusion?

any thoughts?

Reply:
maybe add a general note after the list that: `some validations may be lifted in the future (for instance if the DaemonSet mode supports ServiceMonitor resources)`?